examples/webgl_lights_directional.html

```diff
@@ -0,0 +1,444 @@
+
+					if ( typeof value === 'string' ) {
+
+						value = value.replace( '#', '0x' );
+
+					}
+
+					dirLight.color.setHex( value );
+
+				} );
+
+				lightf.add( datal, 'intensity', 0, 1).onChange( function ( value ) {
+
+					dirLight.intensity = value;
+
+				} );
+
+				var shadowf = gui.addFolder( 'Shadow' );
+				shadowf.open();
+
+				var datas = {
+					size: dirLight.shadow.mapSize.width,
+					left: dirLight.shadow.camera.left,
+					right: dirLight.shadow.camera.right,
+					top: dirLight.shadow.camera.top,
+					bottom: dirLight.shadow.camera.bottom,
+					near: dirLight.shadow.camera.near,
+					far: dirLight.shadow.camera.far,
+				};
+
+				shadowf.add( datas, 'size', {
+					'32': 32,
+					'64': 64,
+					'128': 128,
+					'256': 256,
+					'512': 512,
+					'1024': 1024,
+					'2048': 2048
+				} ).onChange( function ( value ) {
+
+					dirLight.shadow.mapSize.width = dirLight.shadow.mapSize.height = value;
+
+				} );
+
+				shadowf.add( datas, 'left', -250, -10 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.shadow.camera.left = value;
+					setShadowCamera();
+
+				} );
+
+				shadowf.add( datas, 'right', 10, 250 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.shadow.camera.right = value;
+					setShadowCamera();
+
+				} );
+
+				shadowf.add( datas, 'top', 10, 250 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.shadow.camera.top = value;
+					setShadowCamera();
+
+				} );
+
+				shadowf.add( datas, 'bottom', -250, -10 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.shadow.camera.bottom = value;
+					setShadowCamera();
+
+				} );
+
+				shadowf.add( datas, 'near', 0, 150 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.shadow.camera.near = value;
+					setShadowCamera();
+
+				} );
+
+				shadowf.add( datas, 'far', 10, 250 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.shadow.camera.far = value;
+					setShadowCamera();
+
+				} );
+
+			}
+
+			function addTargetGui() {
+
+				var targetf = gui.addFolder( 'Target' );
+				targetf.open();
+
+				var datat = {
+					x: dirLight.target.position.x,
+					y: dirLight.target.position.y,
+					z: dirLight.target.position.z
+				};
+
+				targetf.add( datat, 'x', -25, 25 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.target.position.setX( value );
+					dirLightHelper.update();
+
+				} );
+
+				targetf.add( datat, 'y', -25, 25 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.target.position.setY( value );
+					dirLightHelper.update();
+
+				} );
+
+				targetf.add( datat, 'z', -25, 25 ).step( 1 ).onChange( function ( value ) {
+
+					dirLight.target.position.setZ( value );
+					dirLightHelper.update();
+
+				} );
+
+			}
+
+			function setShadowCamera() {
+
+				if( shadowCameraHelper ) {
+
+					scene.remove( shadowCameraHelper );
+
+				}
+				shadowCameraHelper = new THREE.CameraHelper( dirLight.shadow.camera );

```

Comment:
Calling this every time the GUI is modified is causing hundreds of geometries to be allocated on the GPU.

Type `renderer.info` into the console.

I expect you only need one shadow camera helper.

Reply:
calling `shadowCameraHelper.update()` didn´t change the camera lines, so in order to show the helper changes I choose to create the camera helper everytime. I don't like it either, so If there is a way to avoid this, tell me and I will gladly change it.

Reply:
Please try to track it down. Users copy these examples.

Reply:
See note below.